src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs

```diff
@@ -47,7 +47,8 @@ public string TargetPath
             {
                 if (String.Compare(_targetPath, value, StringComparison.OrdinalIgnoreCase) != 0)
                 {
-                    _targetPath = value;
+                    // Always use forward slash in package.
+                    _targetPath = value.Replace("\\", "/");
```

Comment:
We should have a unit test for this 

1. assign `TargetPath`
1. get and validate:
    1. `TargetPath`
    1. `TargetFramework`
    1. `EffectivePath`